src/http_link.c

```diff
@@ -150,14 +154,28 @@ int http_link_write(uv_link_t *link, uv_link_t *source, const uv_buf_t bufs[], u
                         response->body_size, &resp, &resp_size);
         }
       }
-      else
+      else // Gzip initialized
       {
-        context->request.enable_compression = false;
+        compress_data(response, resp, nread, &resp, &resp_size);
       }
     }
-    else // Gzip initialized
+
+    if (context->first_response)
     {
-      compress_data(response, resp, nread, &resp, &resp_size);
+      context->first_response = false;
+      double timeDiff = (uv_hrtime() - context->request_time) / 1000000.0;
+      struct tm *timeinfo;
+      timeinfo = localtime(&context->request_timestamp);
+      char timeString[256];
+      strftime(timeString, sizeof(timeString), "%d/%b/%Y:%T %z", timeinfo);
+      // IP - [response time] - [date time] "GET url http" HTTP_STATUS_NUM host
```

Comment:
can we somehow detect if the request happened on `http://` or `https://` and prefix `host` with that?
you will also need to know this later when implementing SSL functionality.

Reply:
I'm sure we can detect port number, but currently don't know if "https" will be reflected somewhere else.

Reply:
I think this is not related to port number but to protocol instead. You can set https port number to whatever you want just as you can set http port number to whatever you want, it is not mandatory to be 443 or 80.

I think in SSL case also headers are encrypted so you can detect case of http/https with that. 
Leave that for now but remember to come back later to apply protocol here when you will work on SSL.